common/tools/rasr.py

```diff
@@ -0,0 +1,33 @@
+from typing import Optional
+
+from sisyphus import tk
+
+from i6_core.tools.compile import MakeJob
+from i6_core.tools.git import CloneGitRepositoryJob
+
+
+def compile_rasr_binaries_i6mode(
+    branch: Optional[str] = None,
+    commit: Optional[str] = None,
+    rasr_git_repository: str = "https://github.com/rwth-i6/rasr",
+) -> tk.Path:
+    """
+    Compile RASR for i6-internal usage
+
+    :param branch: specify a specific branch
+    :param commit: specify a specific commit
+    :param rasr_git_repository: where to clone RASR from, usually does not need to be altered
+    :return: path to the binary folder
+    """
+    rasr_repo = CloneGitRepositoryJob(
+        rasr_git_repository, branch=branch, commit=commit
+    ).out_repository
+    make_job = MakeJob(
+        folder=rasr_repo,
+        make_sequence=["build", "install"],
+        configure_opts=["--i6"],
+        num_processes=8,
+        link_outputs={"binaries": "arch/linux-x86_64-standard/"},
+    )
+    make_job.rqmt["mem"] = 8
+    return make_job.out_links["binaries"]
```

Comment:
could we return the `bin_path` and the `arch`??? since we are using these also in combination?

Reply:
we could.. but this function has `arch` as input parameter, so whoever calls it already knows about it.
Otherwise as far as defaults go, both functions (this and the set binaries) have the same default, so setting both to `None` (or not setting) also works.

Reply:
? I dont see `arch` as input parameter.. but that would also be fine for me.. 
but with a different rasr branch we might get a different `arch` so maybe hard coding this is not optimal?!

Reply:
> I dont see `arch` as input parameter..

Sorry, I must have been hallucinating things.
Ok, but I also think, that the current github rasr configure script can only output the default arch.

Nevertheless we could definitively have it as a returned option and/or parameter..

Reply:
> Ok, but I also think, that the current github rasr configure script can only output the default arch.

I thought you can checkout a branch from RASR repo and in the branch you can then have anon default `arch` param... not 100% sure though...

I think having `arch` as a param to the function would allow us the most flexibility..

Reply:
thanks for the explanation :)